src/commands/remove-demo.ts

```diff
@@ -0,0 +1,98 @@
+  run: async (toolbox: GluegunToolbox) => {
+    const { parameters, patching, filesystem } = toolbox
+
+    const CWD = process.cwd()
+    const TARGET_DIR = parameters.first ?? CWD
+    p()
+    p(`Removing demo code from '${TARGET_DIR}'`)
+
+    const getAllFilePaths = createGetAllFilePaths(filesystem)
+    const paths = getAllFilePaths(TARGET_DIR)
+
+    // Go through every file path and handle the operation for each demo comment
+    const demoCommentResults = await Promise.allSettled(
+      paths.map(async (path) => {
+        const { exists, update } = patching
+
+        const comments: CommentType[] = []
+
+        if (await exists(path, demo.CommentType.REMOVE_FILE)) {
+          filesystem.remove(path)
+          comments.push(demo.CommentType.REMOVE_FILE)
+          return { path, comments }
+        }
+
+        if (await exists(path, demo.CommentType.REMOVE_CURRENT_LINE)) {
+          await update(path, demo.removeCurrentLine)
+          comments.push(demo.CommentType.REMOVE_CURRENT_LINE)
+        }
+
+        if (await exists(path, demo.CommentType.REMOVE_NEXT_LINE)) {
+          await update(path, demo.removeNextLine)
+          comments.push(demo.CommentType.REMOVE_NEXT_LINE)
+        }
+
+        if (
+          (await exists(path, demo.CommentType.REMOVE_BLOCK_START)) &&
+          (await exists(path, demo.CommentType.REMOVE_BLOCK_END))
+        ) {
+          await update(path, demo.removeBlock)
+          comments.push(demo.CommentType.REMOVE_BLOCK_START, demo.CommentType.REMOVE_BLOCK_END)
+        }
+
+        return { path, comments }
+      }),
+    )
+
+    // Handle the results of the demo comment operations
+    demoCommentResults
+      // Sort the results by the path in alphabetical order
+      .sort((a, b) => {
+        if (a.status === "fulfilled" && b.status === "fulfilled") {
+          return a.value.path.localeCompare(b.value.path)
+        }
+        return 0
+      })
+      .forEach((result) => {
+        // Log any rejected results as warnings
+        if (result.status === "rejected") {
+          warning(result.reason)
+          return
+        }
+
+        // Log any fulfilled results that have comments
+        const { path, comments } = result.value
+        if (comments.length > 0) {
+          p(`Found ${comments.map((c) => `'${c}'`).join(", ")} in ${path}`)
+        }
+      })
+
+    // Recurisvely remove any empty directories in TARGET_DIR
+    const removeEmptyDirs = (dir: string) => {
+      const files = filesystem.list(dir)
+      if (files.length === 0) {
+        filesystem.remove(dir)
+        p(`Removed empty directory '${dir}'`)
+        return
+      }
+      files.forEach((file) => {
+        const filePath = `${dir}/${file}`
+        if (filesystem.isDirectory(filePath)) {
+          removeEmptyDirs(filePath)
+        }
+      })
+    }
+    removeEmptyDirs(TARGET_DIR)
+
+    p(`Done removing demo code from '${TARGET_DIR}'`)
```

Comment:
We discussed the idea of having a `@demo remove-folder` directive, but I think until it becomes a big issue, we can just add `// @demo remove-file` and then check to see if all the files are deleted in a folder

Reply:
I like this solution, Josh.

Reply:
Thank you! :)